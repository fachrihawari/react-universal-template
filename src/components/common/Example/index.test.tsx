import React from 'react';
import { render } from '@testing-library/react-native';

import WebComponent from './index.web';
import AndroidComponent from './index.android';
import IosComponent from './index.ios';

test('Web component rendered successfully', () => {
  const { queryByText, toJSON } = render(<WebComponent />);
  expect(toJSON()).toMatchSnapshot();
  expect(queryByText('Example on Web')).not.toBeNull();
});

test('Android component rendered successfully', () => {
  const { queryByText, toJSON } = render(<AndroidComponent />);
  expect(toJSON()).toMatchSnapshot();
  expect(queryByText('Example on Android')).not.toBeNull();
});

test('iOS component rendered successfully', () => {
  const { queryByText, toJSON } = render(<IosComponent />);
  expect(toJSON()).toMatchSnapshot();
  expect(queryByText('Example on iOS')).not.toBeNull();
});
