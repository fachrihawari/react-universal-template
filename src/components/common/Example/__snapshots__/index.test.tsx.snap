// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android component rendered successfully 1`] = `
<Text>
  Example on Android
</Text>
`;

exports[`Web component rendered successfully 1`] = `
<Text>
  Example on Web
</Text>
`;

exports[`iOS component rendered successfully 1`] = `
<Text>
  Example on iOS
</Text>
`;
